src/ts/shared/classes/ForeignKeyDbTransformer.ts

```diff
@@ -14,22 +14,22 @@ export type CompoundForeignKeyDefinition = PositionForeignKeyDefinition & {
 
 export interface ForeignKeyDbDefinition {
 	filename: string | string[];
-	entityIds: string | string[];
+	entityId: string;
 	entityName: string;
 }
 
 class ForeignKeyDbTransformer {
 	private foreignKeyDefinition: ForeignKeyDefinition | CompoundForeignKeyDefinition | null;
 
 	public constructor(
-		foreignKeyDefinition: ForeignKeyDefinition | CompoundForeignKeyDefinition | null,
+		foreignKeyDefinition: ForeignKeyDefinition | CompoundForeignKeyDefinition | null = null,
 	) {
 		this.foreignKeyDefinition = foreignKeyDefinition;
 	}
 
 	/** Transform schema foreign key definitions to the format required by the db server */
-	public transform = (value: string | null): ForeignKeyDbDefinition[] => {
-		if (value == null || this.foreignKeyDefinition == null) {
+	public transform = (value: unknown | null): ForeignKeyDbDefinition[] => {
```

Comment:
Since any value could possibly be passed to this method, I had to loosen the type of the parameter.